spec/rubocop/cop/rspec/context_wording_spec.rb

```diff
@@ -0,0 +1,62 @@
+RSpec.describe RuboCop::Cop::RSpec::ContextWording, :config do
+  subject(:cop) { described_class.new(config) }
+
+  let(:cop_config) { { 'Prefixes' => %w[when with] } }
+
+  it 'skips describe blocks' do
+    expect_no_offenses(<<-RUBY)
+      describe 'the display name not present' do
+      end
+    RUBY
+  end
+
+  it 'finds context without `when` at the beginning' do
+    expect_offense(<<-RUBY)
+      context 'the display name not present' do
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Start context description with 'when', or 'with'.
+      end
+    RUBY
+  end
+
+  it 'finds shared_context without `when` at the beginning' do
+    expect_offense(<<-RUBY)
+      shared_context 'the display name not present' do
+                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Start context description with 'when', or 'with'.
+      end
+    RUBY
+  end
+
+  it "skips descriptions beginning with 'when'" do
+    expect_no_offenses(<<-RUBY)
+      context 'when the display name is not present' do
+      end
+    RUBY
+  end
+
+  it 'finds context without separate `when` at the beginning' do
+    expect_offense(<<-RUBY)
+      context 'whenever you do' do
+              ^^^^^^^^^^^^^^^^^ Start context description with 'when', or 'with'.
+      end
+    RUBY
+  end
+
+  context 'when configured' do
```

Comment:
This context applies to all of the specs. But perhaps have a "with default configuration" context, a "with changed configuration" context, and a "with no configuration".

Reply:
I've structured the spec file in a way that root context represents "default configuration" examples, and a "configured cop" context with the same examples as in root context, but different expectations showing how differently configured cop behaves compared to default config.

I can't figure out who would anyone use the cop with empty configuration, e.g. deliberately putting an empty prefix list in config file, they will be getting weird error messages, and that unhappy path is not something I feel we need to add examples for.

Reply:
> I can't figure out who would anyone use the cop with empty configuration … that unhappy path is not something I feel we need to add examples for.

You are absolutely right.